### DISCUSSION TITLE
How to disable camera during the stream

### DISCUSSION THREAD

**Discussion by HaikuOezu** [HaishinKit/HaishinKit.swift#1031] [Q&A] [answered]
Hi guys
I've searched far and wide but I can't find out how to disable/black out the camera mid stream... basically think muting but for video.
Is there a way to "detach" the video mid session or do I have to do something like overlaying a solid black texture over it?

**Comment by shogo4405** [comment, marked as answer]
Please try `rtmpStream.attachCamera(nil)`.

**Reply by HaikuOezu** [reply]
Awesome! Thank you

**Comment by padi-dev-anhntn** [comment]
@shogo4405 こんにちは、私たちは `rtmpStream.attachCamera(nil)` を使ってカメラを無効にしてみましたが、その後、`rtmpStream?.attachCamera(AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: cameraPosition))` を使ってカメラを再度有効にしようとしたところ、うまくいきませんでした。

**Reply by padi-dev-anhntn** [reply]
私たちは `attachCamera` の後に `rtmpConnection?.connect()` と `rtmpStream?.publish()` を試しましたが、もうまく動作しませんでした。

**Reply by padi-dev-anhntn** [reply]
Hello @HaikuOezu , we tried to disable the camera using `rtmpStream.attachCamera(nil)`, but when we tried to enable the camera again using `rtmpStream?.attachCamera(AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: cameraPosition))`, it didn't work. 
We also tried `rtmpConnection?.connect() ` and `rtmpStream?.publish() `after `attachCamera`, but it still didn't work.

**Reply by shogo4405** [reply]
質問ありがとう。技術的な質問にこたえるのやめちゃったんだよね。自身で修正してくれよな。

**Reply by padi-dev-anhntn** [reply]
It’s sad, but I respect that. Thank you.
